### DISCUSSION TITLE
@container queries possible in Lit?

### DISCUSSION THREAD

**Discussion by frmwrk-development** [lit/lit#4793] [General]
I am building a small component with Lit.
The code is based on the LitElement javascript starterkit

It uses a few simple CSS rules:
`static get styles() {
    return css`
      :host {
        display: block;
        container-type: inline-size;
        container-name: host;
      }

...

      @container host (min-width: 600px) {
        .toolbar {
          flex-direction: row;
        }
      }
    `;`

In development mode the container query is working nicely.

However when I build my component, the `@container ... ` parts gets stripped out.

I am not sure where and why this is stripped? Could be done by Lit, or by Rollup.
Anybody with any experience in this area?

**Comment by frmwrk-development** [comment]
For future reference:
Yes, this is possible. In my case the rollup-plugin-minify-html-literals tried to minify my CSS, stripping out my `@container` queries.
I worked around it by disabling CSS minifying.
The proper solution would probably be to update the [Clean-css](https://www.npmjs.com/package/clean-css#constructor-options) configuration.

```    minifyHTML({
      // https://www.npmjs.com/package/minify-html-literals#options
      options: {
        shouldMinifyCSS(template) {
          // Disable minifying CSS, it breaks container queries:
          return false;
        }
      },

    }),```

**Reply by justinfagnani** [reply]
Glad you figured it out. Lit itself doesn't do anything with CSS, so it's up to the browser and tools to keep it working.
